### DISCUSSION TITLE
Why yaml for codegen?

### DISCUSSION THREAD

**Discussion by chimon2000** [serverpod/serverpod#596] [Q&A] [answered]
I was wondering why Serverpod decided to use YAML for the codegen surrounding database communication & serialization, rather than decorators?

**Comment by vlidholt** [comment, marked as answer]
That code was written a long time ago. But it still provides a very clean interface (compared to many popular serialization packages), which makes the code you write easy to read and very compact. That being said, it is also possible to use Serverpod with e.g. Freezed or your own serialization code.
